### ISSUE TITLE
BoolArray fails to get

### ISSUE THREAD

**Issue by keller-mark** [manzt/zarrita.js#118]
https://observablehq.com/d/7152024fe1caf825 Using `get()` on a BoolArray seems to not work as expected - the returned value is `{ data: {}, shape: [34406], stride: [1] }`

**Comment by keller-mark** [comment]
Ah I see now the empty object is an iterator so I can do 

```
Array.from((await zarr.get(arr)).data);
```

so perhaps not a bug after all

**Comment by keller-mark** [comment]
Maybe the user should opt-in to getting an iterator

`zarr.getIterator(arr)`

**Comment by manzt** [comment]
Yeah, the idea was to keep the data in a TypedArray-like object for as long as possible (i.e., a strided view of the underlying bytes). But maybe this is more trouble than it's worth if there aren't use cases for keeping the underlying bytes.

This case happens with the string/bool array types, which is why I introduced the `zarr.Array.is` type guard. You could do something like:


```typescript

if (arr.is("string") || arr.is("bool")) {
   data = Array.from((await get(arr)).data);
} else {
   data = (await get(arr)).data;
}

```

but maybe that's still not very ergonomic. We could probably wrap this in a separate API as you suggested, which will coerce the typed arrays into object arrays.

**Comment by manzt** [comment]
A thought I had yesterday. Maybe we could have a type-aware helper for coercing the data types:

```javascript
let { data } = zarr.refineChunk(await get(array), {
   string: ({ data, shape, stride }) => ({ data: Array.from(data), shape, stride }),
   boolean: ({ data, shape, stride }) => ({ data: Array.from(data), shape, stride }),
})
```

Will need to think about it more, but curious to hear your thoughts.

**Comment by manzt** [comment]
I'm going to close this for now. I think in the future release we could consider deprecating our typed arrays for just `Array<boolean>` and `Array<string>`.
